src/tests/ftest/directory_tree.py

```diff
@@ -162,14 +170,32 @@ def _create_needle(self, current_path, current_height):
 
         self._needles_count -= 1
         suffix = "_{:05d}.needle".format(self._needles_count)
-        fd, file_name = tempfile.mkstemp(
-            dir=current_path, prefix=self._needles_prefix, suffix=suffix)
-        os.close(fd)
+        fname = self._mktemp_file(where=current_path, prefix=self._needles_prefix, suffix=suffix)
+        self._needles_paths.append(fname)
+
+    def _mktemp_file(self, where=None, prefix=None, suffix=None):
+        """Create a temporary file.
+        If the file size is 0, the file will be empty.
+        If the file size is greater than 0, the file will be filled with random data.
+        If min and max file size are different, the file size will be a random between min and max.
+        """
 
-        self._needles_paths.append(file_name)
+        fd, fname = tempfile.mkstemp(dir=where, prefix=prefix, suffix=suffix)
+        if self._file_size_min == 0:
+            os.close(fd)
+            return fname
+
+        size = self._file_size_min
+        if self._file_size_max > self._file_size_min:
+            size = random.randrange(self._file_size_min, self._file_size_max)
```

Comment:
Python bandit
> Standard pseudo-random generators are not suitable for security/cryptographic purposes.

I think we can ignore this one like
```suggestion
            size = random.randrange(self._file_size_min, self._file_size_max)  # nosec
```